autofit/optimize/non_linear/nested_sampling/multi_nest.py

```diff
@@ -210,279 +209,148 @@ def __call__(self, cube, ndim, nparams, lnew):
-        """Open the MultiNest "multinestphys_live_points" file, read it and return all physical live point models as
-        a list of 1D lists."""
-        phys_live = open(self.paths.file_phys_live)
 
-        live_points = 0
-        for line in phys_live:
-            live_points += 1
+def weights_from_file_weighted_samples(file_weighted_samples) -> [float]:
+    """Open the file "multinest.txt" and extract the weight values of every accepted live point as a list."""
+    weighted_samples = open(file_weighted_samples)
 
-        phys_live.seek(0)
+    total_samples = 0
+    for line in weighted_samples:
+        total_samples += 1
 
-        phys_live_points = []
+    weighted_samples.seek(0)
 
-        for line in range(live_points):
-            vector = []
-            for param in range(self.model.prior_count + 1):
-                vector.append(float(phys_live.read(28)))
-            phys_live.readline()
-            phys_live_points.append(vector)
+    log_likelihoods = []
 
-        phys_live.close()
+    for line in range(total_samples):
+        weighted_samples.read(4)
+        log_likelihoods.append(float(weighted_samples.read(28)))
+        weighted_samples.readline()
 
-        return phys_live_points
+    weighted_samples.close()
 
-    def phys_live_points_of_param(self, param_index) -> [float]:
-        """Return all parameter values of a given parameter for all of the current physical live points.
+    return log_likelihoods
 
-        These are read from the "multinestphys_live_points" file."""
-        return [point[param_index] for point in self.phys_live_points]
 
-    def read_list_of_results_from_summary_file(self, number_entries, offset) -> [float]:
-        """Read a list of results from the "multinestsummary.txt" file, which stores information on the MulitNest run
-        incuding the most likely and most probable models.
+def total_samples_from_file_resume(file_resume):
+    """Open the file "resume.dat" and extract the total number of samples of the MultiNest analysis
+    (e.g. accepted + rejected)."""
+    resume = open(file_resume)
 
-        This file stores the results as a text file where different columns correspnd to different models."""
+    resume.seek(1)
+    resume.read(19)
+    return int(resume.read(8))
 
-        summary = open(self.paths.file_summary)
-        summary.read(2 + offset * self.model.prior_count)
-        vector = []
-        for param in range(number_entries):
-            vector.append(float(summary.read(28)))
 
-        summary.close()
+def log_evidence_from_file_summary(file_summary, prior_count):
+    """Open the file "multinestsummary.txt" and extract the log evidence of the Multinest analysis.
+
+    Early in the analysis this file may not yet have been created, in which case the log evidence estimate is
+    unavailable and (would be unreliable anyway). In this case, a large negative value is returned."""
+
+    try:
+        summary = open(file_summary)
```

Comment:
Use a with open and inside that execute everything up to returning vector[0], wrapping the whole thing in the try except